src/pivot-table/components/cells/styles.tsx

```diff
@@ -29,3 +38,31 @@ export const StyledHeaderAnchor = styled(Box)({
   left: 0,
   bottom: 0,
 });
+
+export const StyledHeaderCell = styled("div")(() => ({
+  ...baseFlex,
+  flexDirection: "row",
+  overflow: "hidden",
+  position: "relative",
+}));
+
+export const StyledLockIcon = styled("div")(() => ({
+  ...baseFlex,
+  marginRight: CELL_PADDING,
```

Comment:
Consider using `gap: CELL_PADDING` on `StyledHeaderCell` instead, as that will potentially work better once we implement RTL.